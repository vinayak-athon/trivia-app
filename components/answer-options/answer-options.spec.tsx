import React from "react";
import { render, cleanup, act } from "@Test/utils";
import * as ReactRedux from "react-redux";
import { AnswerOptions } from "@Components";

afterAll(cleanup);
jest.mock("next/router", () => ({
  useRouter: () => ({
    query: { id: "1" },
    push: jest.fn(),
  }),
}));

jest.mock("@Server/i18n", () => ({
  useTranslation: () => ({
    t: x => x,
  }),
}));

describe("Components", () => {
  describe("AnswerOptions", () => {
    const dispatch = jest.fn();
    jest
      .spyOn(ReactRedux, "useDispatch")
      .mockImplementation(() => jest.fn().mockReturnValue({ dispatch }));
    const { container } = render(<AnswerOptions />);
    it("should match snapshot", async () => {
      expect(container).toMatchSnapshot();
    });
  });
});
