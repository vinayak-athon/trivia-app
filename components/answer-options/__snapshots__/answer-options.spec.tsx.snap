// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components AnswerOptions should match snapshot 1`] = `
.c1 {
  cursor: pointer;
  color: #000000;
  background: #f9da2e;
  border-radius: 5px;
  padding: 5px 10px;
  font-family: Arial;
  font-weight: 700;
  font-size: 25px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  text-transform: uppercase;
}

.c0 {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: 10px;
}

.c2 {
  color: #ffffff;
  font-size: 1rem;
  border: none;
  margin: 0 auto;
  width: 100%;
  background: #0062cc;
}

.c3 {
  color: #ffffff;
  font-size: 1rem;
  border: none;
  margin: 0 auto;
  width: 100%;
  background: red;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1 c2 option__true"
    >
      common:True
    </button>
    <button
      class="c1 c3 option__false"
    >
      common:False
    </button>
  </div>
</div>
`;
